dev/SpecUpdater.php

```diff
@@ -31,6 +31,8 @@ public static function update(Event $event): void
                 foreach ($enumFile->fqn::cases() as $existingCase) {
                     if ($enumFile->hasCaseWithValue($existingCase->value) === false) {
                         $enumFile->addCase(new EnumCase($existingCase->name, $existingCase->value, [], true));
+                    } else {
```

Comment:
This part of the code adds any cases that were not found in the source code back to the enum. For example:

Enum A at one point contained case B and C.

Now we update enum A from the source, and case B has disappeared. We can't simply drop it from the spec if someone has it hardcoded like this: A::B, so instead we readd it here and mark it as deprecated (see the last parameter that is set to 'true'). This way, we can release the changes in a minor release without having to go up a major release every time a enum case disappears from the source.

Does that make sense?

Reply:
To make it even more clear: we are iterating over all cases in an enum, and check if we have found it in the enum cases we pulled from the source. In this case, 99% of all the assertions for this if statement will end up in the else statements as only items that have been removed between spec updates will go into the if branch.

Reply:
> Does that make sense?

Yes.

I hope `hasCaseWithValue` will never drop **new** cases!